cli/module_reload.go

```diff
@@ -0,0 +1,108 @@
+package cli
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	apppb "go.viam.com/api/app/v1"
+
+	rdkConfig "go.viam.com/rdk/config"
+	"go.viam.com/rdk/logging"
+)
+
+// ModuleMap is a type alias to indicate where a map represents a module config.
+// We don't convert to rdkConfig.Module because it can get out of date with what's in the db.
+// Using maps directly also saves a lot of high-maintenance ser/des work.
+type ModuleMap map[string]interface{}
+
+// configureModule is the configuration step of module reloading. Returns (needsRestart, error).
+func configureModule(vc *viamClient, manifest *moduleManifest, part *apppb.RobotPart) (bool, error) {
+	logger := logging.Global()
+	if manifest == nil {
+		return false, fmt.Errorf("reconfiguration requires valid manifest json passed to --%s", moduleFlagPath)
+	}
+	partMap := part.RobotConfig.AsMap()
+	modules, err := mapOver(
+		partMap["modules"].([]interface{}),
+		func(raw interface{}) (ModuleMap, error) { return ModuleMap(raw.(map[string]interface{})), nil },
+	)
```

Comment:
```suggestion
	modules, err := mapOver(
		partMap["modules"].([]any),
		func(raw any) (ModuleMap, error) { return ModuleMap(raw.(map[string]any)), nil },
	)
```

This is a total matter of opinion, but I find `any` so much easier to read and understand and it is fewer characters. The go stdlib has transitioned entirely to any.

Reply:
I learned go before `any` and my muscle memory is stale -- will switch over. (I like the new way better too)

Reply:
2833c8ec converted `interface{}` to `any` (go 1.18 ftw)